In [9]:
import warnings
warnings.filterwarnings("ignore")

# Inicialización

Importamos las librerías que nos servirán para poder procesar, visualizar y explorar nuestros datos de manera efectiva, la librería `pandas`, `numpy`, `json`, `gzip`.


In [10]:
import pandas as pd
import numpy as np
import json
import gzip

%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


# Extracción de los datos Steam Games

Extraer datos del archivo JSON "steam_games.json.gz", convertirlos en un DataFrame utilizando Pandas y explorar su contenido inicial.


In [11]:
# Ruta al archivo comprimido
file_path = 'steam_games.json.gz'

# Lista para almacenar los datos
data_list = []

# Descomprime el archivo y lee línea por línea
with gzip.open(file_path, 'rt', encoding='utf-8') as file:
    for line in file:
        # Carga la línea como JSON
        data = json.loads(line)
        data_list.append(data)

# Crea un DataFrame a partir de la lista de datos
data_games = pd.DataFrame(data_list)

In [12]:
data_games

publisher                                 genres  \
0                   NaN                                    NaN   
1                   NaN                                    NaN   
2                   NaN                                    NaN   
3                   NaN                                    NaN   
4                   NaN                                    NaN   
...                 ...                                    ...   
120440  Ghost_RUS Games  [Casual, Indie, Simulation, Strategy]   
120441           Sacada              [Casual, Indie, Strategy]   
120442     Laush Studio            [Indie, Racing, Simulation]   
120443         SIXNAILS                        [Casual, Indie]   
120444              NaN                                    NaN   

                        app_name                     title  \
0                            NaN                       NaN   
1                            NaN                       NaN   
2                            NaN                       NaN   
3                            NaN                       NaN   
4                            NaN                       NaN   
...                          ...                       ...   
120440            Colony On Mars            Colony On Mars   
120441  LOGistICAL: South Africa  LOGistICAL: South Africa   
120442             Russian Roads             Russian Roads   
120443       EXIT 2 - Directions       EXIT 2 - Directions   
120444               Maze Run VR                       NaN   

                                                      url release_date  \
0                                                     NaN          NaN   
1                                                     NaN          NaN   
2                                                     NaN          NaN   
3                                                     NaN          NaN   
4                                                     NaN          NaN   
...                                                   ...          ...   
120440  http://store.steampowered.com/app/773640/Colon...   2018-01-04   
120441  http://store.steampowered.com/app/733530/LOGis...   2018-01-04   
120442  http://store.steampowered.com/app/610660/Russi...   2018-01-04   
120443  http://store.steampowered.com/app/658870/EXIT_...   2017-09-02   
120444  http://store.steampowered.com/app/681550/Maze_...          NaN   

                                                     tags  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
120440              [Strategy, Indie, Casual, Simulation]   
120441                          [Strategy, Indie, Casual]   
120442                        [Indie, Simulation, Racing]   
120443  [Indie, Casual, Puzzle, Singleplayer, Atmosphe...   
120444  [Early Access, Adventure, Indie, Action, Simul...   

                                              reviews_url  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
120440  http://steamcommunity.com/app/773640/reviews/?...   
120441  http://steamcommunity.com/app/733530/reviews/?...   
120442  http://steamcommunity.com/app/610660/reviews/?...   
120443  http://steamcommunity.com/app/658870/reviews/?...   
120444  http://steamcommunity.com/app/681550/reviews/?...   

                                                    specs price early_access  \
0                                        

# Dataset

Para explorar nuestra base de datos vamos a llamar al atributo shape para saber el número total de filas y columnas, posteriormente utilizaremos los métodos head() y tail() para realizar un primer análisis de datos y familiarizarnos con nuestro dataframe. El dataset contiene los siguientes campos:

**publisher:** Empresa publicadora del contenido.

**genres:** Genero del juego. (lista de uno o mas géneros por registro)

**app_name:** Nombre del juego.

**title:** Titulo del contenido.

**url:** URL de publicación del contenido.

**release_date:** Fecha de lanzamiento.

**tags:** Etiqueta del contenido. (lista de uno o mas etiquetas por registro)

**reviews_url:** Reviews de contenido

**specs:** Especificaciones (lista con una o mas especificaciones por registro).

**price:**  Precio del videojuego.

**early_access:** Acceso temprano (Booleano).

**id:** Identificador único del juego.

**developer:** Desarrollador del contenido.

In [13]:
data_games.shape

(120445, 13)

In [14]:
data_games.head(10)

publisher genres app_name title  url release_date tags reviews_url specs  \
0       NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   NaN   
1       NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   NaN   
2       NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   NaN   
3       NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   NaN   
4       NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   NaN   
5       NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   NaN   
6       NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   NaN   
7       NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   NaN   
8       NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   NaN   
9       NaN    NaN      NaN   NaN  NaN          NaN  NaN         NaN   NaN   

  price early_access   id developer  
0   NaN          NaN  NaN       NaN  
1   NaN          NaN  NaN       NaN  
2   NaN          NaN  NaN       NaN  
3   NaN          NaN  NaN       NaN  
4   NaN          NaN  NaN       NaN  
5   NaN          NaN  NaN       NaN  
6   NaN          NaN  NaN       NaN  
7   NaN          NaN  NaN       NaN  
8   NaN          NaN  NaN       NaN  
9   NaN          NaN  NaN       NaN

In [15]:
data_games.tail(10)

publisher                                 genres  \
120435  Retro Army Limited             [Action, Adventure, Indie]   
120436  OrtiGames/OrtiSoft                        [Casual, Indie]   
120437              INGAME                           [Indie, RPG]   
120438            Riviysky                        [Casual, Indie]   
120439     Bidoniera Games     [Action, Adventure, Casual, Indie]   
120440     Ghost_RUS Games  [Casual, Indie, Simulation, Strategy]   
120441              Sacada              [Casual, Indie, Strategy]   
120442        Laush Studio            [Indie, Racing, Simulation]   
120443            SIXNAILS                        [Casual, Indie]   
120444                 NaN                                    NaN   

                        app_name                     title  \
120435      The spy who shot me™      The spy who shot me™   
120436            Raining blocks            Raining blocks   
120437                   Bravium                   Bravium   
120438                     BAE 2                     BAE 2   
120439              Kebab it Up!              Kebab it Up!   
120440            Colony On Mars            Colony On Mars   
120441  LOGistICAL: South Africa  LOGistICAL: South Africa   
120442             Russian Roads             Russian Roads   
120443       EXIT 2 - Directions       EXIT 2 - Directions   
120444               Maze Run VR                       NaN   

                                                      url release_date  \
120435  http://store.steampowered.com/app/771810/The_s...   2018-10-01   
120436  http://store.steampowered.com/app/767590/Raini...   2018-01-04   
120437  http://store.steampowered.com/app/747320/Bravium/   2018-01-04   
120438    http://store.steampowered.com/app/769330/BAE_2/   2018-01-04   
120439  http://store.steampowered.com/app/745400/Kebab...   2018-01-04   
120440  http://store.steampowered.com/app/773640/Colon...   2018-01-04   
120441  http://store.steampowered.com/app/733530/LOGis...   2018-01-04   
120442  http://store.steampowered.com/app/610660/Russi...   2018-01-04   
120443  http://store.steampowered.com/app/658870/EXIT_...   2017-09-02   
120444  http://store.steampowered.com/app/681550/Maze_...          NaN   

                                                     tags  \
120435                         [Action, Adventure, Indie]   
120436                                    [Indie, Casual]   
120437                [Indie, RPG, Puzzle, Tower Defense]   
120438                                    [Indie, Casual]   
120439        [Action, Indie, Casual, Violent, Adventure]   
120440              [Strategy, Indie, Casual, Simulation]   
120441                          [Strategy, Indie, Casual]   
120442                        [Indie, Simulation, Racing]   
120443  [Indie, Casual, Puzzle, Singleplayer, Atmosphe...   
120444  [Early Access, Adventure, Indie, Action, Simul...   

                                              reviews_url  \
120435  http://steamcommunity.com/app/771810/reviews/?...   
120436  http://steamcommunity.com/app/767590/reviews/?...   
120437  http://steamcommunity.com/app/747320/reviews/?...   
120438  http://steamcommunity.com/app/769330/reviews/?...   
120439  http://steamcommunity.com/app/745400/reviews/?...   
120440  http://steamcommunity.com/app/773640/reviews/?...   
120441  http://steamcommunity.com/app/733530/reviews/?...   
120442  http://steamcommunity.com/app/610660/reviews/?...   
120443  http://steamcommunity.com/app/658870/reviews/?...   
120444  http://steamcommunity.com/app/681550/reviews/?...   

                                                    specs  price early_access  \
120435                [Single-player, Captions available]    NaN        False   
120436  [Single-player, Shared/Split Screen, Steam Ach...   0.99        False   
120437  [Single-player, Steam Achievements, Steam Lead...  14.99        False   
120438                                    [Single-player]    NaN        False   
120439   [

En nuestro DataFrame, inicialmente registramos un total de 120445 filas y 13 columnas, hemos identificado que algunas filas contienen valores vacíos en todos los registros, y también hemos encontrado algunos valores nulos en otras áreas del conjunto de datos. 

# Preparación y Limpieza de los Datos

Antes de analizar y modelar los datos, es esencial asegurarnos de que estén limpios y bien preparados. Esto implica tratar valores faltantes, eliminar duplicados, corregir errores y convertir datos en formatos adecuados. Además, es fundamental revisar cada variable de manera inicial para detectar problemas y realizar las correcciones necesarias. Este proceso garantiza que los datos sean confiables y coherentes para análisis posteriores.

## Eliminación de Registros Vacíos

En una primera vista, identificamos filas que contienen valores vacíos en todos los registros. Vamos a eliminar estas filas para asegurar que trabajemos con datos completos y significativos.


In [16]:
# Eliminar filas con valores nulos en todo el registro
data_games = data_games.dropna(how='all').reset_index(drop=True)

In [17]:
data_games.shape

(32135, 13)

hemos eliminado estas filas que contenían todos los datos nulos. El resultado es un DataFrame más limpio y eficiente, con un total de 32,135 filas y 13 columnas.

cantidad de valores nulos por variable en el DataFrame `data_games`.

In [18]:
data_games.isnull().sum()

publisher       8052
genres          3283
app_name           2
title           2050
url                0
release_date    2067
tags             163
reviews_url        2
specs            670
price           1377
early_access       0
id                 2
developer       3299
dtype: int64

# Conocer estructura de algunas de nuestras variables

In [19]:
unique_values = data_games['publisher'].unique()
num_unique_values = len(unique_values)
print(f'Variable: Publisher, Valores Únicos: {num_unique_values}')
print('Valores Únicos:', unique_values)
print('\n')

Variable: Publisher, Valores Únicos: 8240
Valores Únicos: ['Kotoshiro' 'Making Fun, Inc.' 'Poolians.com' ... 'OrtiGames/OrtiSoft'
 'INGAME' 'Bidoniera Games']




In [20]:
unique_values = data_games['price'].unique()
num_unique_values = len(unique_values)
print(f'Variable: price, Valores Únicos: {num_unique_values}')
print('Valores Únicos:', unique_values)
print('\n')

Variable: price, Valores Únicos: 163
Valores Únicos: [4.99 'Free To Play' 'Free to Play' 0.99 2.99 3.99 9.99 18.99 29.99 nan
 'Free' 10.99 1.59 14.99 1.99 59.99 8.99 6.99 7.99 39.99 19.99 7.49 12.99
 5.99 2.49 15.99 1.25 24.99 17.99 61.99 3.49 11.99 13.99 'Free Demo'
 'Play for Free!' 34.99 74.76 1.49 32.99 99.99 14.95 69.99 16.99 79.99
 49.99 5.0 44.99 13.98 29.96 119.99 109.99 149.99 771.71 'Install Now'
 21.99 89.99 'Play WARMACHINE: Tactics Demo' 0.98 139.92 4.29 64.99
 'Free Mod' 54.99 74.99 'Install Theme' 0.89 'Third-party' 0.5 'Play Now'
 299.99 1.29 3.0 15.0 5.49 23.99 49.0 20.99 10.93 1.39
 'Free HITMAN™ Holiday Pack' 36.99 4.49 2.0 4.0 9.0 234.99 1.95 1.5 199.0
 189.0 6.66 27.99 10.49 129.99 179.0 26.99 399.99 31.99 399.0 20.0 40.0
 3.33 199.99 22.99 320.0 38.85 71.7 59.95 995.0 27.49 3.39 6.0 19.95
 499.99 16.06 4.68 131.4 44.98 202.76 1.0 2.3 0.95 172.24 249.99 2.97
 10.96 10.0 30.0 2.66 6.48 19.29 11.15 18.9 2.89 'Play the Demo' 99.0
 87.94 599.0 8.98 9.69 0.49 9.98 9.95 

In [21]:
unique_values = data_games['early_access'].unique()
num_unique_values = len(unique_values)
print(f'Variable: early_access, Valores Únicos: {num_unique_values}')
print('Valores Únicos:', unique_values)
print('\n')

Variable: early_access, Valores Únicos: 2
Valores Únicos: [False True]




In [22]:
data_games['genres']

0            [Action, Casual, Indie, Simulation, Strategy]
1                     [Free to Play, Indie, RPG, Strategy]
2        [Casual, Free to Play, Indie, Simulation, Sports]
3                              [Action, Adventure, Casual]
4                                                      NaN
                               ...                        
32130                [Casual, Indie, Simulation, Strategy]
32131                            [Casual, Indie, Strategy]
32132                          [Indie, Racing, Simulation]
32133                                      [Casual, Indie]
32134                                                  NaN
Name: genres, Length: 32135, dtype: object

In [23]:
data_games['tags']

0            [Strategy, Action, Indie, Casual, Simulation]
1        [Free to Play, Strategy, Indie, RPG, Card Game...
2        [Free to Play, Simulation, Sports, Casual, Ind...
3                              [Action, Adventure, Casual]
4                          [Action, Indie, Casual, Sports]
                               ...                        
32130                [Strategy, Indie, Casual, Simulation]
32131                            [Strategy, Indie, Casual]
32132                          [Indie, Simulation, Racing]
32133    [Indie, Casual, Puzzle, Singleplayer, Atmosphe...
32134    [Early Access, Adventure, Indie, Action, Simul...
Name: tags, Length: 32135, dtype: object

In [24]:
data_games['specs']

0                                          [Single-player]
1        [Single-player, Multi-player, Online Multi-Pla...
2        [Single-player, Multi-player, Online Multi-Pla...
3                                          [Single-player]
4        [Single-player, Full controller support, HTC V...
                               ...                        
32130                  [Single-player, Steam Achievements]
32131    [Single-player, Steam Achievements, Steam Clou...
32132    [Single-player, Steam Achievements, Steam Trad...
32133     [Single-player, Steam Achievements, Steam Cloud]
32134    [Single-player, Stats, Steam Leaderboards, HTC...
Name: specs, Length: 32135, dtype: object

# Verificación de Datos Duplicados

Llevaremos a cabo una verificación de datos duplicados en nuestra columna 'id' del DataFrame. El objetivo es determinar si existen registros duplicados en base a este identificador único. 


In [25]:
duplicated_games = data_games.loc[data_games['id'].duplicated(keep=False)]
if not duplicated_games.empty:
    print("Registros duplicados")
else:
    print("No se encontraron registros duplicados en id. Total: 0")
duplicated_games

Registros duplicados


publisher               genres  \
74                                                   NaN                  NaN   
13894                                 Bethesda Softworks             [Action]   
14573                                 Bethesda Softworks             [Action]   
30961  Warner Bros. Interactive Entertainment, Feral ...  [Action, Adventure]   

                                             app_name  \
74                                                NaN   
13894                Wolfenstein II: The New Colossus   
14573                Wolfenstein II: The New Colossus   
30961  Batman: Arkham City - Game of the Year Edition   

                                                title  \
74                                                NaN   
13894                Wolfenstein II: The New Colossus   
14573                Wolfenstein II: The New Colossus   
30961  Batman: Arkham City - Game of the Year Edition   

                                                     url release_date  \
74                        http://store.steampowered.com/          NaN   
13894          http://store.steampowered.com/app/612880/   2017-10-26   
14573  http://store.steampowered.com/app/612880/Wolfe...   2017-10-26   
30961           http://store.steampowered.com/app/200260   2012-09-07   

                                                    tags  \
74                                                   NaN   
13894  [Action, FPS, Gore, Violent, Alternate History...   
14573  [Action, FPS, Gore, Violent, Alternate History...   
30961  [Action, Open World, Batman, Adventure, Stealt...   

                                             reviews_url  \
74                                                   NaN   
13894  http://steamcommunity.com/app/612880/reviews/?...   
14573  http://steamcommunity.com/app/612880/reviews/?...   
30961                                                NaN   

                                                   specs  price early_access  \
74                                                   NaN  19.99        False   
13894  [Single-player, Steam Achievements, Full contr...  59.99        False   
14573  [Single-player, Steam Achievements, Full contr...  59.99        False   
30961  [Single-player, Steam Achievements, Steam Trad...  19.99        False   

           id                                   developer  
74        NaN                                         NaN  
13894  612880                               Machine Games  
14573  612880                               Machine Games  
30961     NaN  Rocksteady Studios,Feral Interactive (Mac)

Podemos observar que se han detectado dos casos de datos duplicados en la columna 'id'. En el primer caso, el registro numero 13894 y 14573, el 'id' 612880 es idéntico en ambas instancias. Esto sugiere que uno de los registros duplicados puede eliminarse sin afectar la integridad de los datos.

En el segundo caso, se observa que uno de los 'id' es NaN (valor nulo) y la primera aparición presenta más valores NaN que la segunda aparición. Esto puede requerir una revisión adicional para determinar si uno de los registros debe mantenerse o si ambos son redundantes.

Procedemos a observar la columna 'developer', que nos proporcionará información sobre el desarrollador o creador del juego.

In [26]:
data_games[data_games['developer']=='Rocksteady Studios,Feral Interactive (Mac)']

publisher               genres  \
1068   Warner Bros. Interactive Entertainment, Feral ...  [Action, Adventure]   
30961  Warner Bros. Interactive Entertainment, Feral ...  [Action, Adventure]   
31617  Warner Bros. Interactive Entertainment, Feral ...  [Action, Adventure]   

                                             app_name  \
1068   Batman: Arkham City - Game of the Year Edition   
30961  Batman: Arkham City - Game of the Year Edition   
31617  Batman: Arkham Asylum Game of the Year Edition   

                                                title  \
1068   Batman: Arkham City - Game of the Year Edition   
30961  Batman: Arkham City - Game of the Year Edition   
31617  Batman: Arkham Asylum Game of the Year Edition   

                                                     url release_date  \
1068   http://store.steampowered.com/app/200260/Batma...   2012-09-07   
30961           http://store.steampowered.com/app/200260   2012-09-07   
31617  http://store.steampowered.com/app/35140/Batman...   2010-03-26   

                                                    tags  \
1068   [Action, Open World, Batman, Adventure, Stealt...   
30961  [Action, Open World, Batman, Adventure, Stealt...   
31617  [Action, Batman, Stealth, Adventure, Third Per...   

                                             reviews_url  \
1068   http://steamcommunity.com/app/200260/reviews/?...   
30961                                                NaN   
31617  http://steamcommunity.com/app/35140/reviews/?b...   

                                                   specs  price early_access  \
1068   [Single-player, Steam Achievements, Steam Trad...  19.99        False   
30961  [Single-player, Steam Achievements, Steam Trad...  19.99        False   
31617  [Single-player, Steam Achievements, Steam Trad...  19.99        False   

           id                                   developer  
1068   200260  Rocksteady Studios,Feral Interactive (Mac)  
30961     NaN  Rocksteady Studios,Feral Interactive (Mac)  
31617   35140  Rocksteady Studios,Feral Interactive (Mac)

Al observar la columna 'developer', que nos proporciona información sobre el desarrollador o creador del juego, notamos que el registro sin un valor 'id' (el 'id' 200260) está completo y no presenta faltantes de información.

Por lo tanto, en este escenario, se sugiere eliminar ambas filas duplicadas, ya que el registro sin 'id' completo no afecta la integridad de los datos y puede considerarse redundante.

In [27]:
# Eliminar filas con los índices 74, 14573, 30961

data_games = data_games.drop([ 74, 14573, 30961])

In [28]:
# Nos aseguramos que no hay datos duplicados en nuestra columna 'id' del DataFrame


duplicated_games = data_games.loc[data_games['id'].duplicated(keep=False)]
if not duplicated_games.empty:
    print("Registros duplicados")
else:
    print("No se encontraron registros duplicados en id. Total: 0")


No se encontraron registros duplicados en id. Total: 0


# Eliminación de Columnas no Relevantes 

En esta fase del proceso, nos encontramos con la necesidad de simplificar nuestro DataFrame para centrarnos en las variables clave que serán utilizadas en consultas futuras a la API y en la construcción de nuestro modelo de recomendación. Por lo tanto, hemos decidido eliminar las columnas `url`, `tags`, `reviews_url` y `specs` en esta primera instancia.

Es importante destacar que esta eliminación no se realiza debido a que estas variables carezcan de importancia, sino más bien porque esta información no será utilizada en las consultas a la API ni en la generación del modelo de recomendación. Este enfoque contribuirá a simplificar nuestro conjunto de datos y nos permitirá concentrarnos en las variables que son esenciales para nuestras tareas de análisis y recomendación.


In [29]:
data_games = data_games.drop(['tags', 'specs', 'url', 'reviews_url'], axis=1)
data_games.head(10)

publisher                                             genres  \
0               Kotoshiro      [Action, Casual, Indie, Simulation, Strategy]   
1        Making Fun, Inc.               [Free to Play, Indie, RPG, Strategy]   
2            Poolians.com  [Casual, Free to Play, Indie, Simulation, Sports]   
3                    彼岸领域                        [Action, Adventure, Casual]   
4                     NaN                                                NaN   
5     Trickjump Games Ltd                    [Action, Adventure, Simulation]   
6                     NaN          [Free to Play, Indie, Simulation, Sports]   
7  Poppermost Productions          [Free to Play, Indie, Simulation, Sports]   
8  Poppermost Productions          [Free to Play, Indie, Simulation, Sports]   
9               RewindApp                [Casual, Indie, Racing, Simulation]   

                        app_name                          title release_date  \
0            Lost Summoner Kitty            Lost Summoner Kitty   2018-01-04   
1                      Ironbound                      Ironbound   2018-01-04   
2        Real Pool 3D - Poolians        Real Pool 3D - Poolians   2017-07-24   
3                        弹炸人2222                        弹炸人2222   2017-12-07   
4                  Log Challenge                            NaN          NaN   
5          Battle Royale Trainer          Battle Royale Trainer   2018-01-04   
6   SNOW - All Access Basic Pass   SNOW - All Access Basic Pass   2018-01-04   
7     SNOW - All Access Pro Pass     SNOW - All Access Pro Pass   2018-01-04   
8  SNOW - All Access Legend Pass  SNOW - All Access Legend Pass   2018-01-04   
9                           Race                           Race   2018-01-04   

          price early_access      id               developer  
0          4.99        False  761140               Kotoshiro  
1  Free To Play        False  643980        Secret Level SRL  
2  Free to Play        False  670290            Poolians.com  
3          0.99        False  767400                    彼岸领域  
4          2.99        False  773570                     NaN  
5          3.99        False  772540     Trickjump Games Ltd  
6          9.99        False  774276  Poppermost Productions  
7         18.99        False  774277  Poppermost Productions  
8         29.99        False  774278  Poppermost Productions  
9           NaN        False  768800               RewindApp

# Transformación de Columnas y Gestión de Valores Nulos

Continuando con nuestra preparación de datos, nos enfocamos en las columnas `publisher`, `app_name`, `title` y `developer`. Al revisar estas columnas, observamos la presencia de valores nulos en algunos registros. Para asegurarnos de que las variables sean consistentes y manejar adecuadamente los datos faltantes, decidimos completar los valores nulos con la etiqueta 'Dato no disponible'.


In [30]:
# Reemplazo de valores nulos en las columnas 'publisher', 'app_name', 'title' y 'developer'
data_games['publisher'].fillna('Dato no disponible', inplace=True)
data_games['app_name'].fillna('Dato no disponible', inplace=True)
data_games['title'].fillna('Dato no disponible', inplace=True)
data_games['developer'].fillna('Dato no disponible', inplace=True)


In [35]:
data_games.head()

publisher      genres             app_name                title  \
0  Kotoshiro      Action  Lost Summoner Kitty  Lost Summoner Kitty   
0  Kotoshiro      Casual  Lost Summoner Kitty  Lost Summoner Kitty   
0  Kotoshiro       Indie  Lost Summoner Kitty  Lost Summoner Kitty   
0  Kotoshiro  Simulation  Lost Summoner Kitty  Lost Summoner Kitty   
0  Kotoshiro    Strategy  Lost Summoner Kitty  Lost Summoner Kitty   

  release_date price early_access      id  developer  
0   2018-01-04  4.99        False  761140  Kotoshiro  
0   2018-01-04  4.99        False  761140  Kotoshiro  
0   2018-01-04  4.99        False  761140  Kotoshiro  
0   2018-01-04  4.99        False  761140  Kotoshiro  
0   2018-01-04  4.99        False  761140  Kotoshiro



# Transformación de la Columna 'genres'

A partir de la columna 'genres', aplicaremos una transformación específica para descomponer las listas de géneros de videojuegos en registros individuales. 

In [36]:
# Usar explode para descomponer la columna 'genres'
data_games = data_games.explode('genres')

# Eliminar filas con valores nulos en la columna 'genres'
data_games = data_games.dropna(subset=['genres'])


In [37]:
data_games.shape

(71551, 9)

In [38]:
data_games[data_games['genres']=='Accounting']

publisher      genres                           app_name  \
5219   Individual Software  Accounting         Quicken Legal Business Pro   
8499   Individual Software  Accounting        Quicken WillMaker Plus 2017   
11751       MING-CHIEN LEE  Accounting                            MyMoney   
12296  Individual Software  Accounting        Quicken WillMaker Plus 2018   
15811             Stardock  Accounting                       Multiplicity   
17192  Individual Software  Accounting  Professor Teaches QuickBooks 2017   
24387  Individual Software  Accounting        Quicken WillMaker Plus 2016   

                                   title release_date  price early_access  \
5219          Quicken Legal Business Pro   2015-10-15  44.99        False   
8499         Quicken WillMaker Plus 2017   2016-11-08  54.99        False   
11751                            MyMoney   2017-09-30   Free        False   
12296        Quicken WillMaker Plus 2018   2017-11-10  69.99        False   
15811                       Multiplicity   2017-09-07  19.99        False   
17192  Professor Teaches QuickBooks 2017   2017-04-03  14.99        False   
24387        Quicken WillMaker Plus 2016   2015-10-15  59.99        False   

           id            developer  
5219   411350                 Nolo  
8499   555810                 Nolo  
11751  677310       MING-CHIEN LEE  
12296  732710                 Nolo  
15811  620040             Stardock  
17192  623590  Individual Software  
24387  411340                 Nolo

# Transformación de la Columna 'release_date'
A partir de la columna 'release_date', extraeremos el año de lanzamiento y crearemos una nueva columna para almacenar este dato. En caso de que la fecha esté ausente, se marcará como "Dato no disponible".


In [39]:
# Convertir la columna 'release_date' al tipo datetime
data_games['release_date'] = pd.to_datetime(data_games['release_date'], errors='coerce')

# Extraer el año y crear la columna 'release_year', asignando -1 a los valores no válidos
data_games['release_year'] = data_games['release_date'].dt.year.fillna(-1).astype(int)

# Reemplazar los valores -1 por "Dato no disponible" en la columna 'release_year'
data_games['release_year'] = data_games['release_year'].replace(-1, "Dato no disponible")

# Eliminar la columna 'release_date' original
data_games.drop('release_date', axis=1, inplace=True)

# Transformación de la Columna 'price'

En la columna 'price' Reemplazaremos los valores de texto, como promociones o juegos gratuitos, por 0 para facilitar futuros análisis. También imputaremos los valores nulos como 0 para mantener coherencia en el conjunto de datos.

In [40]:
# Reemplazar cualquier valor de texto por 0, formatear a dos decimales y asignar 0 a los valores nulos en la columna 'price'
data_games['price'] = data_games['price'].apply(lambda x: 0 if isinstance(x, str) else round(float(x), 2)).fillna(0)


In [41]:
data_games.head()

publisher      genres             app_name                title  price  \
0  Kotoshiro      Action  Lost Summoner Kitty  Lost Summoner Kitty   4.99   
0  Kotoshiro      Casual  Lost Summoner Kitty  Lost Summoner Kitty   4.99   
0  Kotoshiro       Indie  Lost Summoner Kitty  Lost Summoner Kitty   4.99   
0  Kotoshiro  Simulation  Lost Summoner Kitty  Lost Summoner Kitty   4.99   
0  Kotoshiro    Strategy  Lost Summoner Kitty  Lost Summoner Kitty   4.99   

  early_access      id  developer release_year  
0        False  761140  Kotoshiro         2018  
0        False  761140  Kotoshiro         2018  
0        False  761140  Kotoshiro         2018  
0        False  761140  Kotoshiro         2018  
0        False  761140  Kotoshiro         2018

In [44]:
# Reemplazar 'Dato no disponible' con NaN
#data_items['release_year'] = data_items['release_year'].replace('Dato no disponible', np.nan)

# Eliminar filas con NaN en la columna 'release_year'
#data_items.dropna(subset=['release_year'], inplace=True)

In [45]:
data_games.isnull().sum()

publisher       0
genres          0
app_name        0
title           0
price           0
early_access    0
id              0
developer       0
release_year    0
dtype: int64

In [46]:
data_games.shape

(71551, 9)

# Carga del Conjunto de Datos Transformado

En esta etapa, cargaremos el conjunto de datos transformado y depurado, el cual hemos nombrado `"data_games_cleaned.csv"`. Este archivo refleja nuestro conjunto de datos preparado y optimizado para análisis y modelado de datos

In [286]:
# Especifica el nombre del archivo
nombre_del_archivo = 'data_games_cleaned.csv'

# Guarda el DataFrame en el archivo CSV
data_games.to_csv(nombre_del_archivo, index=False, encoding='utf-8')
print(f'Se ha guardado el archivo {nombre_del_archivo} en la misma carpeta.')



Se ha guardado el archivo data_games_cleaned.csv en la misma carpeta.
